airbyte-integrations/connectors/source-stripe/source_stripe/availability_strategy.py

```diff
@@ -0,0 +1,27 @@
+import logging
+from typing import Optional, Tuple
+
+from airbyte_cdk.sources import Source
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http.availability_strategy import HttpAvailabilityStrategy
+
+
+class StripeSubStreamAvailabilityStrategy(HttpAvailabilityStrategy):
+    def check_availability(self, stream: Stream, logger: logging.Logger, source: Optional[Source]) -> Tuple[bool, Optional[str]]:
+        """Traverse through all the parents of a given stream and run availability strategy on each of them"""
+        try:
+            current_stream, parent_stream = stream, getattr(stream, "parent")
+        except AttributeError:
+            return super().check_availability(stream, logger, source)
+        # 1. Instantiate a parent stream
+        # 2. Run availability strategy of the parent stream if specified
+        # 3. Run availability strategy of the current stream
```

Comment:
I think the optimized version of the code is clear enough to not need the commenting anymore - wdyt?

Reply:
done